indico/modules/events/timetable/controllers/legacy.py

```diff
@@ -133,3 +137,43 @@ def _process(self):
             if start_dt:
                 entries.append(schedule_contribution(contribution, start_dt=start_dt))
         return jsonify(entries=[serialize_entry_update(x) for x in entries])
+
+
+class RHLegacyTimetableEntryMove(RHManageTimetableEntryBase):
+    """Moves a TimetableEntry into a Session or top-level timetable
+    """
+
+    def _process_POST(self):
+        current_day = as_utc(datetime.strptime(request.form.get('current_day'), '%Y%m%d')).date()
+        return jsonify_template('events/timetable/move_timetable_entry.html', event=self.event_new,
+                                top_level_entries=self._get_session_timetable_entries(),
+                                current_day=current_day, timetable_entry=self.timetable_entry,
+                                parent_entry=self.timetable_entry.parent)
+
+    def _process_PATCH(self):
+        serializer = TimetableSerializer(True)
+        with track_time_changes():
+            rv = self._move_entry(request.json, serializer)
+        rv['entry'] = serializer.serialize_timetable_entry(self.timetable_entry)
+        return jsonify_data(**rv)
```

Comment:
You have to use `serialize_entry_update`, otherwise you need to modify the legacy JS, as you did, potentially breaking stuff. If you need, modify the serialization method to match the expected information on the client-side.